processor-steps/src/main/java/boringyuri/processor/common/steps/ext/Elements.kt

```diff
@@ -0,0 +1,184 @@
+    val type = type.typeName
+    val ensureNonNull = !type.isPrimitive && defaultValue != null
+
+    return FieldSpec.builder(type, fieldName, Modifier.PRIVATE)
+        .addAnnotations(
+            annotationHandler.toAnnotationSpec(
+                this.type,
+                getAllAnnotations(),
+                ensureNonNull
+            )
+        )
+        .build()
+}
+
+fun XMethodElement.createFieldSpec(
+    paramName: String,
+    defaultValue: String?,
+    annotationHandler: AnnotationHandler
+): FieldSpec {
+    val fieldName = FIELD_PREFIX + StringUtils.capitalize(paramName)
+    val type = returnType.typeName
+    val ensureNonNull = !type.isPrimitive && defaultValue != null
+
+    return FieldSpec.builder(type, fieldName, Modifier.PRIVATE)
+        .addAnnotations(
+            annotationHandler.toAnnotationSpec(
+                returnType,
+                getAllAnnotations(),
+                ensureNonNull
+            )
+        )
+        .build()
+}
+
+fun XVariableElement.createMethodSignature(
+    defaultValue: String?,
+    annotationHandler: AnnotationHandler
+): MethodSpec.Builder {
+    val paramName = name
+    val paramType = type.typeName
+    val methodName = buildGetterName(paramName, paramType)
+    val ensureNonNull = !paramType.isPrimitive && defaultValue != null
+
+    return MethodSpec.methodBuilder(methodName)
+        .addModifiers(Modifier.PUBLIC)
+        .addAnnotations(
+            annotationHandler.toAnnotationSpec(
+                type,
+                getAllAnnotations(),
+                ensureNonNull
+            )
+        )
+        .returns(paramType)
+}
+
+fun XMethodElement.createMethodSignature(
+    defaultValue: String?,
+    annotationHandler: AnnotationHandler
+): MethodSpec.Builder {
+    val paramType = returnType.typeName
+    val ensureNonNull = !paramType.isPrimitive && defaultValue != null
+
+    return MethodSpec.methodBuilder(name)
+        .addModifiers(Modifier.PUBLIC)
+        .addAnnotations(
+            annotationHandler.toAnnotationSpec(returnType, getAllAnnotations(), ensureNonNull)
+        )
+        .addParameters(parameters.map { it.createParamSpec(annotationHandler) })
+        .returns(paramType)
+}
+
+fun XElement.findTypeAdapter(): XType? {
+    val adapter = getAnnotationValueAsType<TypeAdapter>()
+    if (adapter != null) {
+        return adapter
+    }
+    val type = when (this) {
+        is XVariableElement -> type
+        is XMethodElement -> returnType
+        else -> null
+    }
+    return type?.accept(TypeAdapterVisitor(), null)
+        ?.getAsType("value")
+}
+
+fun XAnnotated.requireTypeAdapter(): XType {
+    return requireAnnotationValueAsType<TypeAdapter>()
+}
+
+private class TypeAdapterVisitor : TypeVisitor<XAnnotationBox<TypeAdapter>?, Unit?> {
+    override fun visit(type: XType, param: Unit?): XAnnotationBox<TypeAdapter>? {
+        return type.typeElement?.getAnnotation(TypeAdapter::class)
```

Comment:
nit: I think here you may use your newly created inline function with reified type.
```suggestion
        return type.typeElement?.getAnnotation<TypeAdapter>()
```

Reply:
Not really. To be able to get `XType` from annotation value I need to work with `XAnnotationBox` here

Reply:
alright, then keep it as it is.